snapcraft/internal/meta.py

```diff
@@ -177,10 +176,8 @@ def write_snap_directory(self):
                     destination = os.path.join(dst_dir, asset)
 
                     # First, verify that the hook is actually executable
-                    if (origin == 'hooks' and
-                            not os.stat(source).st_mode & stat.S_IEXEC):
-                        raise CommandError(
-                            'hook {!r} is not executable'.format(asset))
+                    if origin == 'hooks':
+                        _validate_hook(source)
```

Comment:
I was really talking about the overall migration function, something like `_migrate_into_snapdir(path, checker=None)` where the checker could validate the hook when the `path='hooks'`. But yeah, this is fine.